src/components/DateInput.jsx

```diff
@@ -93,7 +93,7 @@ class DateInput extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (!this.props.displayValue && nextProps.displayValue) {
+    if (nextProps.displayValue) {
```

Comment:
The previous logic says “when a display value was not present, and you start passing one, clear the date string” - your new logic says “every time you render with a display value, even if it’s the same one as before, reset the date string”. I suspect this isn’t your intention?

Reply:
Actually, it is.
This is how you are defining `value` in the `render` method.
```javascript
    const value = displayValue || dateString || '';
```
So, basically, you are using `dateString` only if `displayValue` is empty or null.

The only thing that needs to be added to the condition is `!this.state.dateString` to prevent unnecessary `setState`.

So it will look like this:

```javascript
    if (!this.state.dateString && nextProps.displayValue) {
      this.setState({
        dateString: '',
      });
    }
```